src/history/hs.c

```diff
@@ -728,11 +710,24 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_PAGE *page, WT_MULTI *multi)
             if (prev_upd != NULL && prev_upd->start_ts < upd->start_ts)
                 enable_reverse_modify = false;
 
+            /* Find the first update without timestamp. */
+            if (first_non_ts_upd == NULL && upd->start_ts == WT_TS_NONE) {
+                first_non_ts_upd = upd;
+            } else if (first_non_ts_upd != NULL && upd->start_ts != WT_TS_NONE) {
+                /*
+                 * Don't insert updates with timestamps after updates without timestamps to the
+                 * history store.
+                 */
+                F_SET(upd, WT_UPDATE_MASKED_BY_NON_TS_UPDATE);
```

Comment:
Is it still fine that we're not caring about transaction ids? Not sure what the expectation is now.

Reply:
It's not correct theoretically, we rely on mongodb not reading anything before the mixed mode operation. If they do, then we have a problem.